src/components/scenes/ConfirmScene.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react'
+import { View } from 'react-native'
+import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
+
+import { useHandler } from '../../hooks/useHandler'
+import s from '../../locales/strings'
+import { NavigationProp, RouteProp } from '../../types/routerTypes'
+import { SceneWrapper } from '../common/SceneWrapper'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { EdgeText } from '../themed/EdgeText'
+import { MainButton } from '../themed/MainButton'
+import { SafeSlider } from '../themed/SafeSlider'
+import { SceneHeader } from '../themed/SceneHeader'
+import { Tile } from '../tiles/Tile'
+interface Props {
+  navigation: NavigationProp<'confirmScene'>
+  route: RouteProp<'confirmScene'>
+}
+
+export interface ConfirmSceneParams {
+  titleText: string
+  bodyText: string
+  infoTiles?: Array<{ label: string; value: string }>
+  onConfirm: (resetSlider: () => void) => void
+  onBack?: () => void
+}
+
+const ConfirmComponent = (props: Props) => {
+  const { navigation, route } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const { titleText, bodyText, infoTiles, onConfirm, onBack } = route.params
+
+  const renderInfoTiles = () => {
+    if (!infoTiles || !infoTiles.length) return null
+    return infoTiles.map(({ label, value }) => <Tile key={label} type="static" title={label} body={value} />)
+  }
+
+  const handleSliderComplete = useHandler(async (resetSlider: () => void) => {
+    onConfirm(resetSlider)
+  })
+
+  const handleBackButton = useHandler(() => {
+    navigation.pop()
+    if (onBack != null) onBack()
+  })
+
+  React.useEffect(() => {
+    return () => {
+      if (onBack != null) onBack()
+    }
+  }, [onBack])
```

Comment:
Something about the navigation upgrade breaks this, but I don't know all the details. We need to make @samholmes aware of this so he can fix it in his upgrade branch.

Reply:
This also exists in the SendScene2 so I assume he'll have to fix it there as well.

Reply:
Right, that's why I know about it. Sam had to come up with a fix there, but I don't remember the details.